echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandler.java

```diff
@@ -63,7 +63,11 @@ public void handle(Event event, Map postedEvent) {
         objectMapper.convertValue(postedEvent, GithubWebhookEvent.class);
 
     event.content.put("hash", githubWebhookEvent.after);
-    event.content.put("branch", githubWebhookEvent.ref.replace("refs/heads/", ""));
+    if (githubWebhookEvent.ref == null) {
+      event.content.put("branch", "");
+    } else {
+      event.content.put("branch", githubWebhookEvent.ref.replace("refs/heads/", ""));
```

Comment:
@dotdotdotpaul this could probably be simplified by using `Optional.of(githubWebhookEvent.ref).orElse("").replace...`.

Reply:
Could you also ensure this is covered in the test?

Reply:
Will add a test.  Not sure the "Optional" thing is that much more "simplified".  IMHO, it kinda clutters the line, which is already pretty long.